### ISSUE TITLE
Errors compiling on Windows with non-C:\ folder names that have spaces

### ISSUE THREAD

**Issue by angelsix** [gbdk-2020/gbdk-2020#506]
Download windows 64 release. Rename Examples folder to "Examples With A Space", now nothing compiles.

Seems no way to get lcc.exe to accept any input files where the name or path to those files has spaces in it, which is incredibly common.

**Comment by bbbbbr** [comment]
Does it work when passing a quoted path to lcc? 
If I recall there is some handling in lcc for path quoting.

**Comment by angelsix** [comment]
No. Nothing makes it work. I noticed its entirely related to if its on a non C:\ drive. Drive D:\ is not a network drive, its just a second hard drive

![image](https://github.com/gbdk-2020/gbdk-2020/assets/4455939/f654e2ea-cc8b-48e9-82ed-fd454aeaedb1)

**Comment by bbbbbr** [comment]
Thanks for the screen shots. 
Can you run that same command with -v as an lcc argument? it will show the build steps as lcc tries to execute them.

I have an idea about what might be going on (the issue doesn't seem to occur on linux, so it'll take some more time to get my testing set up)

**Comment by angelsix** [comment]
![image](https://github.com/gbdk-2020/gbdk-2020/assets/4455939/a9bee730-ea50-44cb-a8d8-af4f9565708b)

**Comment by bbbbbr** [comment]
Thanks

**Comment by bbbbbr** [comment]
Just confirming reproduction of this on a D drive folder with some spaces.

Also noting for future reference this transient cc1 path error that appeared to self correct (?) on the second run on a C drive folder with spaces.
```

c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre>cd examples\gb\template_minimal
c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples\gb\template_minimal>compile.bat
c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples\gb\template_minimal>REM Automatically generated from Makefile

c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples\gb\template_minimal>..\..\..\bin\lcc  -o Example.gb main.c
sdcpp.exe: fatal error: cannot execute 'cc1': CreateProcess: No such file or directory
compilation terminated.
at 1: warning 190: ISO C forbids an empty translation unit
subprocess error 1

c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples\gb\template_minimal>compile.bat
c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples\gb\template_minimal>REM Automatically generated from Makefile

c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples\gb\template_minimal>..\..\..\bin\lcc  -o Example.gb main.c
c:\gb\gbdk_4.2.0\gbdk 4.2.0 pre\examples>
```

**Comment by bbbbbr** [comment]
Had reproduced it earlier, but now it seems to compile without issue. Guess will have to test more and see what comes up.

```
d:\GB Space\examples\gb\colorbar>..\..\..\bin\lcc -Wa-l -Wl-m  -v -c -o colorbar.o colorbar.c
..\..\..\bin\lcc $Id: lcc.c,v 2.0 2023/06/01 23:14:24 gbdk-2020 Exp $
"d:\GB Space\"bin/sdcc -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"d:\GB Space\"lib/gb -DINT_16_BITS -I"d:\GB Space\"include -c colorbar.c -o colorbar.o

d:\GB Space\examples\gb\colorbar>..\..\..\bin\lcc -Wa-l -Wl-m -v -Wm-yC -o colorbar.gb colorbar.o
..\..\..\bin\lcc $Id: lcc.c,v 2.0 2023/06/01 23:14:24 gbdk-2020 Exp $
"d:\GB Space\"bin/sdldgb -n -i -m -g _shadow_OAM=0xC000 -g .STACK=0xE000 -g .refresh_OAM=0xFF80 -b _DATA=0xC0A0 -b _CODE=0x0200 -k "d:\GB Space\"lib/sm83/ -l sm83.lib -k "d:\GB Space\"lib/gb/ -l gb.lib colorbar.ihx "d:\GB Space\"lib/gb/crt0.o colorbar.o
"d:\GB Space\"bin/ihxcheck colorbar.ihx
"d:\GB Space\"bin/makebin -yN -Z -yC colorbar.ihx colorbar.gb

d:\GB Space\examples\gb\colorbar>

d:\GB Space\examples\gb\colorbar>make clean
rm -f *.o *.lst *.map *.gb *.ihx *.sym *.cdb *.adb *.asm

d:\GB Space\examples\gb\colorbar>make
../../../bin/lcc -Wa-l -Wl-m  -c -o colorbar.o colorbar.c
../../../bin/lcc -Wa-l -Wl-m  -Wm-yC -o colorbar.gb colorbar.o

d:\GB Space\examples\gb\colorbar>

d:\GB Space\examples\gb\colorbar>compile.bat

d:\GB Space\examples\gb\colorbar>REM Automatically generated from Makefile

d:\GB Space\examples\gb\colorbar>..\..\..\bin\lcc -Wa-l -Wl-m  -c -o colorbar.o colorbar.c

d:\GB Space\examples\gb\colorbar>..\..\..\bin\lcc -Wa-l -Wl-m  -Wm-yC -o colorbar.gb colorbar.o

d:\GB Space\examples\gb\colorbar>
```

**Comment by Nik4ant** [comment]
Have the same problem as `angelsix`. I did 3 tests:
- gbdk on D: drive, but path has no spaces [Ok]
- gbdk on C: drive, but path contains spaces [Ok]
- gbdk on D: drive, but path contains spaces [Error]

Error in the terminal is the same as posted [above](https://github.com/gbdk-2020/gbdk-2020/issues/506#issuecomment-1546713599) by `angelsix`:
![image](https://github.com/gbdk-2020/gbdk-2020/assets/63895276/b02d209b-6ddd-4351-85cc-083453acc19b)


NOTE: `..\game-src\main.c` contains `..\game-src` part because this build command is generated relative to my python script, which is located in `GB Competition 2023\tools\build.py`. This doesn't result into error if used inside `GB Competition 2023\game-src`, but looks wierd, so I thought it would be a good idea to point that out

**Comment by bbbbbr** [comment]
Would you mind re-running the Error scenario with these flags for lcc? ` -v -Wf--verbose -Wf-Wp-v ` it turns on verbose mode for lcc, the sdcc compiler and the sdcpp preprocessor.

I tried some and remain reproduce the issue on my Windows VM anymore. (tested with GBDK 4.1.1 and the pre-release 4.2.0)

**Comment by bbbbbr** [comment]
Looking at it closer, it's possible this could be a path issue for the preprocessor or calling the preprocessor. Due to that, it may be worth trying the pre-release GBDK 4.2.0 since the compiler package included a lot of updates to the preprocessor version.

You can download a version of the 4.2.0 pre-release here:
(must be logged into github, they require that for some reason for GH actions run artifacts)
https://github.com/gbdk-2020/gbdk-2020/actions/runs/5528502249

**Comment by Nik4ant** [comment]
NOTE: Both commands were executed in `D:\My stuff\projects\GB Competition 2023\tools`
- Command without spaces: `D:\gbdk\bin\lcc -v -Wf--verbose -Wf-Wp-v -o ..\game-builds\game.gb ..\game-src\config.c ..\game-src\main.c`:
```
D:\gbdk\bin\lcc.exe $Id: lcc.c,v 2.0 2022/11/05 04:01:12 gbdk-2020 Exp $
..\game-src\config.c:
"D:\gbdk\"bin/sdcc -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\gbdk\"lib/gb -DINT_16_BITS --verbose -Wp-v -I"D:\gbdk\"include -c ..\game-src\config.c -o C:\Users\Nikita\AppData\Local\Temp/lcc129160.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO -D__SDCC_CHAR_SIGNED -D__SDCC_INT_LONG_REENT -D__SDCC_FLOAT_REENT -D__SDCCCALL=1 -D__SDCC=4_2_2 -D__SDCC_VERSION_MAJOR=4 -D__SDCC_VERSION_MINOR=2 -D__SDCC_VERSION_PATCH=2 -D__SDCC_REVISION=13350 -D__SDCC_sm83 -D__STDC_NO_COMPLEX__=1 -D__STDC_NO_THREADS__=1 -D__STDC_NO_ATOMICS__=1 -D__STDC_NO_VLA__=1 -D__STDC_ISO_10646__=201409L -D__STDC_UTF_16__=1 -D__STDC_UTF_32__=1 -D__SDCC_BITINT_MAXWIDTH=64 -isystem "D:\gbdk\bin\..\include\sm83" -isystem "D:\gbdk\bin\..\include"  "..\game-src\config.c"
sdcc: Generating code...
ignoring nonexistent directory "D:/gbdk/bin/../include/sm83"
#include "..." search starts here:
#include <...> search starts here:
 D:/gbdk/lib/gb
 D:/gbdk/include
 D:/gbdk/bin/../include
End of search list.
..\game-src\config.c:1: warning 190: ISO C forbids an empty source file
sdcc: Calling assembler...
sdcc: sdasgb -pogn -plosgffw "C:\Users\Nikita\AppData\Local\Temp/lcc129160.o" "C:\Users\Nikita\AppData\Local\Temp/lcc129160".asm
..\game-src\main.c:
"D:\gbdk\bin/sdcc" -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\gbdk\lib/gb" -DINT_16_BITS --verbose -Wp-v -I"D:\gbdk\include" -c ..\game-src\main.c -o C:\Users\Nikita\AppData\Local\Temp/lcc129161.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO -D__SDCC_CHAR_SIGNED -D__SDCC_INT_LONG_REENT -D__SDCC_FLOAT_REENT -D__SDCCCALL=1 -D__SDCC=4_2_2 -D__SDCC_VERSION_MAJOR=4 -D__SDCC_VERSION_MINOR=2 -D__SDCC_VERSION_PATCH=2 -D__SDCC_REVISION=13350 -D__SDCC_sm83 -D__STDC_NO_COMPLEX__=1 -D__STDC_NO_THREADS__=1 -D__STDC_NO_ATOMICS__=1 -D__STDC_NO_VLA__=1 -D__STDC_ISO_10646__=201409L -D__STDC_UTF_16__=1 -D__STDC_UTF_32__=1 -D__SDCC_BITINT_MAXWIDTH=64 -isystem "D:\gbdk\bin\..\include\sm83" -isystem "D:\gbdk\bin\..\include"  "..\game-src\main.c"
sdcc: Generating code...
ignoring nonexistent directory "D:/gbdk/bin/../include/sm83"
#include "..." search starts here:
#include <...> search starts here:
 D:/gbdk/lib/gb
 D:/gbdk/include
 D:/gbdk/bin/../include
End of search list.
sdcc: Calling assembler...
sdcc: sdasgb -pogn -plosgffw "C:\Users\Nikita\AppData\Local\Temp/lcc129161.o" "C:\Users\Nikita\AppData\Local\Temp/lcc129161".asm
"D:\gbdk\"bin/sdldgb -n -i -g _shadow_OAM=0xC000 -g .STACK=0xE000 -g .refresh_OAM=0xFF80 -b _DATA=0xC0A0 -b _CODE=0x0200 -k "D:\gbdk\"lib/sm83/ -l sm83.lib -k "D:\gbdk\"lib/gb/ -l gb.lib ..\game-builds\game.ihx "D:\gbdk\"lib/gb/crt0.o C:\Users\Nikita\AppData\Local\Temp/lcc129160.o C:\Users\Nikita\AppData\Local\Temp/lcc129161.o
"D:\gbdk\"bin/ihxcheck ..\game-builds\game.ihx
"D:\gbdk\"bin/makebin -yN -Z ..\game-builds\game.ihx ..\game-builds\game.gb
rm C:\Users\Nikita\AppData\Local\Temp/lcc129161.adb ..\game-builds\game.ihx C:\Users\Nikita\AppData\Local\Temp/lcc129160.o C:\Users\Nikita\AppData\Local\Temp/lcc129160.asm C:\Users\Nikita\AppData\Local\Temp/lcc129160.lst C:\Users\Nikita\AppData\Local\Temp/lcc129160.sym C:\Users\Nikita\AppData\Local\Temp/lcc129160.adb C:\Users\Nikita\AppData\Local\Temp/lcc129161.o C:\Users\Nikita\AppData\Local\Temp/lcc129161.asm C:\Users\Nikita\AppData\Local\Temp/lcc129161.lst C:\Users\Nikita\AppData\Local\Temp/lcc129161.sym
```

- Command with spaces: ` & "D:\My stuff\gbdk\bin\lcc.exe" -v -Wf--verbose -Wf-Wp-v -o ..\game-builds\game.gb ..\game-src\config.c ..\game-src\main.c`
```
D:\My stuff\gbdk\bin\lcc.exe $Id: lcc.c,v 2.0 2022/11/05 04:01:12 gbdk-2020 Exp $
..\game-src\config.c:
"D:\My stuff\gbdk\"bin/sdcc -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\My stuff\gbdk\"lib/gb -DINT_16_BITS --verbose -Wp-v -I"D:\My stuff\gbdk\"include -c ..\game-src\config.c -o C:\Users\Nikita\AppData\Local\Temp/lcc123320.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\My stuff\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO -D__SDCC_CHAR_SIGNED -D__SDCC_INT_LONG_REENT -D__SDCC_FLOAT_REENT -D__SDCCCALL=1 -D__SDCC=4_2_2 -D__SDCC_VERSION_MAJOR=4 -D__SDCC_VERSION_MINOR=2 -D__SDCC_VERSION_PATCH=2 -D__SDCC_REVISION=13350 -D__SDCC_sm83 -D__STDC_NO_COMPLEX__=1 -D__STDC_NO_THREADS__=1 -D__STDC_NO_ATOMICS__=1 -D__STDC_NO_VLA__=1 -D__STDC_ISO_10646__=201409L -D__STDC_UTF_16__=1 -D__STDC_UTF_32__=1 -D__SDCC_BITINT_MAXWIDTH=64 -isystem "D:\My stuff\gbdk\bin\..\include\sm83" -isystem "D:\My stuff\gbdk\bin\..\include"  "..\game-src\config.c"
sdcc: Generating co"D:\My" не является внутренней или внешней
командой, исполняемой программой или пакетным файлом.
de...
at 1: warning 190: ISO C forbids an empty source file
..\game-src\main.c:
"D:\My stuff\gbdk\bin/sdcc" -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\My stuff\gbdk\lib/gb" -DINT_16_BITS --verbose -Wp-v -I"D:\My stuff\gbdk\include" -c ..\game-src\main.c -o C:\Users\Nikita\AppData\Local\Temp/lcc123321.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\My stuff\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO -D__SDCC_CHAR_SIGNED -D__SDCC_INT_LONG_REENT -D__SDCC_FLOAT_REENT -D__SDCCCALL=1 -D__SDCC=4_2_2 -D__SDCC_VERSION_MAJOR=4 -D__SDCC_VERSION_MINOR=2 -D__SDCC_VERSION_PATCH=2 -D__SDCC_REVISION=13350 -D__SDCC_sm83 -D__STDC_NO_COMPLEX__=1 -D__STDC_NO_THREADS__=1 -D__STDC_NO_ATOMICS__=1 -D__STDC_NO_VLA__=1 -D__STDC_ISO_10646__=201409L -D__STDC_UTF_16__=1 -D__STDC_UTF_32__=1 -D__SDCC_BITINT_MAXWIDTH=64 -isystem "D:\My stuff\gbdk\bin\..\include\sm83" -isystem "D:\My stuff\gbdk\bin\..\include"  "..\game-src\main.c"
sdcc: Generating code...
"D:\My" не является внутренней или внешней
командой, исполняемой программой или пакетным файлом.
at 1: warning 190: ISO C forbids an empty source file
rm C:\Users\Nikita\AppData\Local\Temp/lcc123321.adb C:\Users\Nikita\AppData\Local\Temp/lcc123320.o C:\Users\Nikita\AppData\Local\Temp/lcc123320.asm C:\Users\Nikita\AppData\Local\Temp/lcc123320.lst C:\Users\Nikita\AppData\Local\Temp/lcc123320.sym C:\Users\Nikita\AppData\Local\Temp/lcc123320.adb C:\Users\Nikita\AppData\Local\Temp/lcc123321.o C:\Users\Nikita\AppData\Local\Temp/lcc123321.asm C:\Users\Nikita\AppData\Local\Temp/lcc123321.lst C:\Users\Nikita\AppData\Local\Temp/lcc123321.sym
```

**Comment by Nik4ant** [comment]
Error occurs even in `4.2.0` version.


After running: `& "D:\My stuff\gbdk\bin\lcc.exe" -v -Wf--verbose -Wf-Wp-v -o ..\game-builds\game.gb ..\game-src\config.c ..\game-src\main.c`, output is: 
```
D:\My stuff\gbdk\bin\lcc.exe $Id: lcc.c,v 2.0 2023/07/12 06:55:35 gbdk-2020 Exp $
..\game-src\config.c:
"D:\My stuff\gbdk\"bin/sdcc -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\My stuff\gbdk\"lib/gb -DINT_16_BITS --verbose -Wp-v -I"D:\My stuff\gbdk\"include -c ..\game-src\config.c -o C:\Users\Nikita\AppData\Local\Temp/lcc162240.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\My stuff\gbdk\include" --obj-ext=.rel -D__SDCC_STACK_AUTO -D__SDCC_CHAR_SIGNED -D__SDCC_INT_LONG_REENT -D__SDCC_FLOAT_REENT -D__SDCCCALL=1 -D__SDCC=4_3_2 -D__SDCC_VERSION_MAJOR=4 -D__SDCC_VERSION_MINOR=3 -D__SDCC_VERSION_PATCH=2 -D__SDCC_REVISION=14228 -D__SDCC_sm83 -D__STDC_NO_COMPLEX__=1 -D__STDC_NO_THREADS__=1 -D__STDC_NO_ATOMICS__=1 -D__STDC_NO_VLA__=1 -D__STDC_ISO_10646__=201409L -D__SIZEOF_FLOAT__=4 -D__SIZEOF_DOUBLE__=4 -D__SDCC_BITINT_MAXWIDTH=64 -isystem "D:\My stuff\gbdk\bin\..\include\sm83" -isystem "D:\My stuff\gbdk\bin\..\include"  -xc "..\game-src\config.c"
sdcc: Generating code...
"D:\My" не является внутренней или внешней
командой, исполняемой программой или пакетным файлом.
at 1: warning 190: ISO C forbids an empty translation unit
subprocess error 1
..\game-src\main.c:
"D:\My stuff\gbdk\bin/sdcc" -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\My stuff\gbdk\lib/gb" -DINT_16_BITS --verbose -Wp-v -I"D:\My stuff\gbdk\include" -c ..\game-src\main.c -o C:\Users\Nikita\AppData\Local\Temp/lcc162241.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\My stuff\gbdk\include" --obj-ext=.rel -D__SDCC_STACK_AUTO -D__SDCC_CHAR_SIGNED -D__SDCC_INT_LONG_REENT -D__SDCC_FLOAT_REENT -D__SDCCCALL=1 -D__SDCC=4_3_2 -D__SDCC_VERSION_MAJOR=4 -D__SDCC_VERSION_MINOR=3 -D__SDCC_VERSION_PATCH=2 -D__SDCC_REVISION=14228 -D__SDCC_sm83 -D__STDC_NO_COMPLEX__=1 -D__STDC_NO_THREADS__=1 -D__STDC_NO_ATOMICS__=1 -D__STDC_NO_VLA__=1 -D__STDC_ISO_10646__=201409L -D__SIZEOF_FLOAT__=4 -D__SIZEOF_DOUBLE__=4 -D__SDCC_BITINT_MAXWIDTH=64 -isystem "D:\My stuff\gbdk\bin\..\include\sm83" -isystem "D:\My stuff\gbdk\bin\..\include"  -xc "..\game-src\main.c"
sdcc: Generating code...
"D:\My" не является внутренней или внешней
командой, исполняемой программой или пакетным файлом.
at 1: warning 190: ISO C forbids an empty translation unit
subprocess error 1
rm C:\Users\Nikita\AppData\Local\Temp/lcc162241.adb C:\Users\Nikita\AppData\Local\Temp/lcc162240.o C:\Users\Nikita\AppData\Local\Temp/lcc162240.asm C:\Users\Nikita\AppData\Local\Temp/lcc162240.lst C:\Users\Nikita\AppData\Local\Temp/lcc162240.sym C:\Users\Nikita\AppData\Local\Temp/lcc162240.adb C:\Users\Nikita\AppData\Local\Temp/lcc162241.o C:\Users\Nikita\AppData\Local\Temp/lcc162241.asm C:\Users\Nikita\AppData\Local\Temp/lcc162241.lst C:\Users\Nikita\AppData\Local\Temp/lcc162241.sym
```

**Comment by bbbbbr** [comment]
Thanks for generating these. 

It does seem to be when the SDCC compiler is trying to invoke the SDCPP preprocessor. This is a part of the toolchain where we have less direct control (though if we can identify the issue specifically we can either try to do a workaround or ideally a patch/bug report upstream).

I made some changes (reboot, alter how D drive is attached and sourced) and I'm not able to reproduce the issue. So I'll continue looking into it.

**Comment by bbbbbr** [comment]
### tldr; 
This appears to be a bug caused by SDCC behavior + WIndows 8.3 filename default settings for non-C drives. There is a workaround below to enable it for D: drives/etc. 

Probably a bug report should be opened for SDCC, but not sure if there will be an easy code solution. Maybe that Windows handling code is really old and there are modern options now.

### Problem Description
The problem seems to be that SDCC tries to generate a Windows "short filename" (8.3) to work around path quoting acceptance issues. Here with GetShortPathName():
https://sourceforge.net/p/sdcc/code/12359/tree/trunk/sdcc/src/SDCCsystem.c#l165

However in current versions of Windows the 8.3 filenames are usually only enabled by default for the "C" drive, and must be turned on manually for other drives. That's why it generally works on D: drive paths with no spaces - there is no quoting problem that 8.3 would be required to solve.
- https://stackoverflow.com/questions/24068756/short-names-only-works-on-c-drive
- https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-R2-and-2012/ff621566(v=ws.11)?redirectedfrom=MSDN

### Workaround
To query and enable it for D Drive, then make sure the desired folders have matching 8.3 entries:
https://michlstechblog.info/blog/windows-enable-generation-of-8-3-names/
* Run Windows `Command` as administrator
* `fsutil.exe 8dot3name query D:`
  * The volume state is: 1 (8dot3 name creation is disabled). The registry state is: 2 (Per volume setting - the default). Based on the above settings, 8dot3 name creation is disabled on D:
* `fsutil 8dot3name set D: 0`
  * Successfully enabled 8dot3name generation on D:
* `fsutil.exe 8dot3name query D:`
  * The volume state is: 0 (8dot3 name creation is enabled). The registry state is: 2 (Per volume setting - the default). Based on the above settings, 8dot3 name creation is enabled on D:

Next: 
* Only folders created AFTER the setting has been enabled will get 8.3 filenames, renaming folders does NOT appear to generate an 8.3 filename. 
* So you must create a new folder or manually generate an 8.3 filename for an existing folder.
  * To manually generate the 8.3 filename for an existing folder here is an example:
    * Make sure no command prompt and explorer window has that filename / path active
    * You may need to do this for every folder with spaces in the path to the GBDK bin dir (where SDCC and SDCPP are stored) 
    * `D:\>fsutil file setshortname "D:\My stuff" "mystuf~1"`  
    * https://learn.microsoft.com/en-us/archive/blogs/winsdk/getshortpathname-doesnt-return-short-path-name
* You can check it with `dir /X` in it's parent folder to see if it shows a 8.3 filename
 ```>dir d:\ /X
 Volume in drive D is D Drive
 Volume Serial Number is 2413-6770

 Directory of d:\

07/14/2023  10:57 PM    <DIR>                       My stuff renamed  <--- This was created before enabling support
07/15/2023  12:10 AM    <DIR>          MYSTUF~1     My stuff2   <---  This was created after enabling support 
```

### Example of the Difference
When working sdcc generates a Windows "short filename" for calling sdcpp since the C: drive DID have 8.3 filenames turned ON by default.
 (found `-Wf-V` to show additional verbose execution):
```
C:\My stuff\projects\GB Competition 2023\game-src>"C:\My stuff\gbdk\bin\lcc.exe" -v -Wf--verbose -Wf-Wp-v -Wf-V  -o ..\game-builds\game.gb  ..\game-src\main.c
(snip)
...
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"C:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"C:\My stuff\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO (snip)

(THIS LINE BELOW:)
+ C:\MYSTUF~1\gbdk\bin\sdcpp.exe -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"C:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"C:\My stuff\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO (snip)

sdcc: Generating code...
ignoring nonexistent directory "C:/My stuff/gbdk/bin/../include/sm83"
(snip)
```

When failing, it does not generate the short filename since the D: drive had 8.3 filenames turned OFF by default:
```D:\My stuff\projects\GB Competition 2023\game-src> "D:\My stuff\gbdk\bin\lcc.exe" -v -Wf--verbose -Wf-Wp-v -Wf-V -o ..\game-builds\game.gb ..\game-src\config.c ..\game-src\main.c
D:\My stuff\gbdk\bin\lcc.exe $Id: lcc.c,v 2.0 2022/11/05 04:01:12 gbdk-2020 Exp $
D:\My stuff\gbdk\bin\lcc.exe: can't find `..\game-src\config.c'
..\game-src\main.c:
"D:\My stuff\gbdk\"bin/sdcc -msm83 --no-std-crt0 --fsigned-char --use-stdout -D__PORT_sm83 -D__TARGET_gb -Wa-pogn -I"D:\My stuff\gbdk\"lib/gb -DINT_16_BITS --verbose -Wp-v -V -I"D:\My stuff\gbdk\"include -c ..\game-src\main.c -o C:\Users\IEUser\AppData\Local\Temp/lcc46960.o
sdcc: Calling preprocessor...
sdcc: sdcpp -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\My stuff\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO (snip)

(THIS LINE BELOW:)
+ D:\My stuff\gbdk\bin\sdcpp.exe -nostdinc -Wall -std=c11 -D"__PORT_sm83" -D"__TARGET_gb" -I"D:\My stuff\gbdk\lib/gb" -D"INT_16_BITS" -v -I"D:\My stuff\gbdk\include" -obj-ext=.rel -D__SDCC_STACK_AUTO (snip)

sdcc: Generating code...
'D:\My' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by bbbbbr** [comment]
Closing this as `won't-fix` since it's an upstream SDCC problem and there is a workaround. Perhaps we can find a fix for SDCC that will be acceptable.  

I'll make a note of this in the documentation.

**Comment by bbbbbr** [comment]
Other projects seem to have encountered this in some form:
https://github.com/xamarin/xamarin-android/pull/2062
